pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/LookupProxyHandler.java

```diff
@@ -85,6 +87,9 @@ public class LookupProxyHandler {
             .create().register();
     private final Semaphore lookupRequestSemaphore;
 
+    // Maps the topic name from the request to the full topic name
+    private final Map<String, String> topicNameCache = new HashMap<>();
```

Comment:
When multiple connections access the same topic, will it cause more heap memory occupation?

Reply:
Yes. However, once a connection is disconnected, the cache will be garbage collected. In extreme cases, if many consumers subscribe 

How about removing the cache? This conversion is actually very fast. It's only about 2.2x slower than reading from a `TopicName` cache.

```
TopicNameBenchmark.testConvert        thrpt       284.114          ops/s
TopicNameBenchmark.testReadFromCache  thrpt       644.718          ops/s
```

and it can execute 30000*284 operations in 1 second.

<img width="1415" alt="image" src="https://github.com/user-attachments/assets/7a62c381-8102-46e9-9fee-e71077c0563c" />

Reply:
i.e. each operation only spends only 0.12 **MICROSECONDS**. The "space-for-time tradeoff" might not worth here.